doc/Sonic-mclag-hld.md

```diff
@@ -0,0 +1,668 @@
+    - [9.1.1. Add MCLAG configuration](#911-Add-MCLAG-configuration)
+    - [9.1.2. Add Acl_table and Acl_rule_table in app-db](#912-Add-Acltable-and-Aclruletable-in-app-db)
+  - [9.2. Add mclagsyncd process](#92-Add-mclagsyncd-process)
+  - [9.3. CLI design](#93-CLI-design)
+  - [9.4. aclorch changes](#94-aclorch-changes)
+  - [9.5. portmgr changes](#95-portmgr-changes)
+  - [9.6. portsorch changes](#96-portsorch-changes)
+  - [9.7. intfmgr changes](#97-intfmgr-changes)
+  - [9.8. intfsorch changes](#98-intfsorch-changes)
+  - [9.9. routeorch changes](#99-routeorch-changes)
+  - [9.10. fdborch changes](#910-fdborch-changes)
+  - [9.11. vxlanmgr changes](#911-vxlanmgr-changes)
+  - [9.12. vxlanorch changes](#912-vxlanorch-changes)
+  - [9.13. vnetorch changes](#913-vnetorch-changes)
+  - [9.14. warm-reboot consideration](#914-warm-reboot-consideration)
+  - [9.15. teammgr changes](#915-teammgr-changes)
+- [10. Test](#10-Test)
+
+<!-- /TOC -->
+
+# 1. Document History
+
+| Version | Date       | Author                 | Description                                      |
+|---------|------------|------------------------|--------------------------------------------------|
+| v.01    | 07/30/2018 |Jianjun Dong            | Initial version from nephos                      |
+| v.02    | 10/19/2018 |Jianjun Dong, Shine Chen, Jeffrey Zeng | Revised per review meeting with SONIC community  |
+| v.03    | 10/27/2018 |Jeffrey Zeng                 | Minor update to clarify the behavior.            |
+| v.04    | 03/01/2019 |Jianjun Dong, Shine Chen, Jeffrey Zeng | Revised per review meeting with SONIC community  |
+| v.05    | 04/26/2019 |Jianjun Dong                    | Add pull request on chapter 9                    |
+| v.06    | 06/10/2019 |Jianjun Dong, Jeffrey Zeng, Shine Chen | Add L2 forwarding description and more use cases |
+| v.07    | 06/19/2019 |Jianjun Dong, Jeffrey Zeng, Shine Chen | Revised per review meeting with SONIC community  |
+| v1.0    | 07/23/2019 |Jianjun Dong, Jeffrey Zeng, Shine Chen | Revised per review meeting with SONIC community  |
+| v1.01   | 11/27/2019 |Jianjun Dong            | Add ND sync description                          |
+
+# 2. Abbreviations
+
+|**Term** |**Definition**                                    |
+|---------|--------------------------------------------------|
+|MCLAG  |Multi-Chassis Link Aggregation Group              |
+|ICCP  |Inter-Chassis Communication Protocol              |
+|FDB     |Layer-2 (MAC) based forwarding table              |
+
+# 3. Terminologies
+
+![Diagram 3](https://github.com/shine4chen/SONiC/blob/mclag/images/mclag_hld/MCLAG_HLD_3.png)
+Diagram 3
+
+# 4. Design considerations
+
+- MCLAG domain consists of only two systems.
+- Each system only join one MC-LAG domain
+- Supports Known Unicast and BUM traffic
+- L3 interface on MLAG ports will have vMAC generated from VRRP algorithm using the same IP address assigned to the L3 LIF (logical interface)；(Not supported currently)
```

Comment:
If VRRP is used, in diagram 6.2, VRRP needs to run on portchannel0001.

Reply:
according to diagram 6.2 the portchannel0001 is a routed interface rather than a switching interface (which belongs to a vlan). in this case which interface can vrrp on peer1/2 communicate on?

Reply:
Sorry, the expression is not accurate enough. We may use VRRP algorithm to generate VMAC instead of VRRP protocol.